### PR TITLE
Feature add api post question

### PR DESCRIPTION
**Pull request by psitronic**
A REST API to post/add question:
- link /api/postQuestion
- all arguments in the request body as:
key: question
value: {"messageText":"Empty",
            "messageTitle":"Hi",
            "messageTags":[],
            "messageAuthor":"5c0108713d25cfdde5a5a4b3"}

where Author is the ObjectID
- if a question posted successfully:
```
{
    "success": true,
    "message": "Question posted",
    "question": {
        "tags": [],
        "answers": [],
        "scores": 0,
        "views": 0,
        "edited": false,
        "_id": "5c1b44e29641e654e5d1d16d",
        "author": "5c0108713d25cfdde5a5a4b3",
        "title": "privet",
        "text": "New way!",
        "date": "2018-12-20T07:29:38.656Z",
        "__v": 0
    }
}
```

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      author: parsedQuestion.messageAuthor,
+      title: parsedQuestion.messageTitle,
+      text: parsedQuestion.messageText,
+      tags: parsedQuestion.messageTags,
+    };
+    Questions.create(newQuestion, (error, questionPosted) => {
+      if (!error) {
+        res.json({
+          success: true,
+          message: 'Question posted'
```
Comment (nikrb) [inline server/routes/api.js]: I think we should pass the new question _id aswell

File: server/routes/api.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const Questions = require('../models/questions');
+const db = require('../models/db');
+
+const api = express.Router();
+
+// post question
+api.post('/postQuestion', (req, res) => {
```
Comment (nikrb) [inline server/routes/api.js]: I think route should be just `/question`

**Review by nikrb — CHANGES_REQUESTED** [review]
works for me.

File: server/routes/api.js (hunk 1 above)
Comment (psitronic) [reply server/routes/api.js]: Thanks for pointing it out. Actually I wanted to return an entire posted document. Now it is fixed

File: server/routes/api.js (hunk 2 above)
Comment (psitronic) [reply server/routes/api.js]: I am not sure, because we also need to get question(s) from the DB, so there should be many routes.

**Review by Francis-Njoku — DISMISSED** [review]

File: server/routes/api.js (hunk 2 above)
Comment (nikrb) [reply server/routes/api.js]: yes, but the requests to question should be `/question`, whether they be get, put, post, delete etc. Atleast that's what I understand by a loosely REST based approach.
Can you give an example of why you would call them e.g. `/postQuestion`, `getQuestion`. Maybe I have the wrong end of the stick?

Comment (psitronic) [reply server/routes/api.js]: Technically, it is ok to use the same `/question` for `get` and `post`, but it is then not clear what exactly it does when you see `/question` get request. Does it request one last question, any question, all questions, etc.? It is just for convenience... However, I am fine with `/question` as well, at least at the moment.

Comment (psitronic) [reply server/routes/api.js]: Fixed

File: server/routes/api.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  if (req.body.question) {
+    const parsedQuestion = JSON.parse(req.body.question);
```
Comment (nikrb) [inline server/routes/api.js:9]: sorry forgot to mention this breaks my test, so have removed it locally. I think this is done by the body parser. How are you testing the route locally?

Comment (psitronic) [reply server/routes/api.js:9]: I use Postman

Comment (nikrb) [reply server/routes/api.js:9]: ok fair enough. I take it the parse was necessary. Interesting I breaks a request from front end. Let's leave it for now and see how things go.

**Comment by nikrb** [comment]
@psitronic sorry dude one last thing and I think we're good to go.
linter objects to db definition in api.js:
```
/home/nik/projects/chingu/voyage7/bears17/server/routes/api.js
  3:7  error  'db' is assigned a value but never used  no-unused-vars
```

**Comment by psitronic** [comment]
> @psitronic sorry dude one last thing and I think we're good to go.
> linter objects to db definition in api.js:
> 
> ```
> /home/nik/projects/chingu/voyage7/bears17/server/routes/api.js
>   3:7  error  'db' is assigned a value but never used  no-unused-vars
> ```

I need to execute the code in db.js in order to connect the DB. Of course it could be done in the api.js or even index.js file but I think it should be in the different file.

**Comment by psitronic** [comment]
Now I moved the DB connection to index.js. It perhaps looks more logical.

**Comment by nikrb** [comment]
@psitronic so sorry dude, messed you about a treat there, many apologies.
I think it's a better idea to have the db conn in a different file, as you said. When I ran it locally, I just removed that db import in index.js and it still worked, hence my comment it could be removed. I didn't intend for 'everything' to be moved into index.js.

I think it's best to leave it as is for now and we can move it around if need be later.

**Review by nikrb — APPROVED** [review]

**Pull request merged**